### PR TITLE
Fix how the expression value_type works

### PR DESCRIPTION
**Pull request by Sutto**
According to #1617 and the tests accompanying, the logic for an `expr`  value type should be "pull the value we're comparing to from a field on the resource."

An example of this is comparing the value of a tag to a name (for instance), or comparing the values of two fields.

Currently, what it actually does is:

1. Use `key` to get the "value" from the resource.
2. Use the returned "value" as an expression on the resource to get the new value.

Where as in 2) we should be using the "sentinel" (aka the value passed to the `value:` field in the filter.

Note that this breaks compatibility for anyone using an older version of the `expr` field - so we probably want to either break it like so (and deal with the fall out), or introduce a new "resource_expr" or similar value type instead.

### CODE REVIEW

File: c7n/filters/core.py [hunk 1]
```diff
@@ -471,11 +471,12 @@ def process_value_type(self, sentinel, value, resource):
...
 
         elif self.vtype == 'expr':
-            return sentinel, self.get_resource_value(value, resource)
+            sentinel = self.get_resource_value(sentinel, resource)
+            return sentinel, value
 
         elif self.vtype == 'integer':
             try:
-                value = int(value.strip())
+                value = giint(value.strip())
```
Comment (kapilt) [inline c7n/filters/core.py]: looks a typo here re giint vs int

Comment (Sutto) [reply c7n/filters/core.py]: yep, definitely - will fix here

**Comment by kapilt** [comment]
thanks for the pr, and looking into this, i'll have to check through some extant policies, but afaics the extant behavior of expr is an aberration that doesn't have a sane semantic afaics. ie. i think its reasonable to to treat this primarily as a bug fix.

**Comment by Sutto** [comment]
@kapilt I've rebased on master, and fixed the failing flake item - will make sure tests pass for this, so it's safe to merge in.

**Comment by kapilt** [comment]
Thanks, there's still some oddity in the test i'm looking at there's a check for sentinel to be a string, but in the test we pass it as None, thats causing it to not reach the test line here and fail out.

**Pull request merged**